core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3483,9 +3492,6 @@ public void run(Reactor session) throws Exception {
                             setSecurityRealm(SecurityRealm.NO_AUTHENTICATION);
                         else
                             setSecurityRealm(new LegacySecurityRealm());
-                    } else {
-                        // force the set to proxy
-                        setSecurityRealm(securityRealm);
```

Comment:
Replaced by the call in `loadConfig()` which is closer to where the `securityRealm` field is set.

Reply:
FTR https://github.com/jenkinsci/jenkins/blob/47c26ade47c51fd705f93a8e97e75a69771cf072/core/src/main/java/jenkins/model/Jenkins.java#L3395-L3410